### PR TITLE
WT-9785 Support basic tiered data source statistics.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: src/cursor/cur_stat.c [hunk 1]
```diff
@@ -745,7 +767,8 @@ __wt_curstat_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *other, c
     if (0) {
 config_err:
         WT_ERR_MSG(session, EINVAL,
-          "cursor's statistics configuration doesn't match the database statistics configuration");
+          "cursor's statistics configuration doesn't match the database statistics "
+          "configuration");
```
Comment (keitharnoldsmith) [inline src/cursor/cur_stat.c:771]: Did you intentionally break this into two strings?

**Review by keitharnoldsmith — APPROVED** [review]
LGTM. One minor comment/question.

File: src/cursor/cur_stat.c (hunk 1 above)
Comment (sueloverso) [reply src/cursor/cur_stat.c:771]: This was from clang format. I did not change anything there myself.

**Comment by sueloverso** [comment]
@ddanderson can you check the change in `hook_tiered.py` as I left change it to the backup project PM-2532. It feels harder to find and notice project tickets in FIXME statements, but there is precedent in that file for it too.

**Comment by sueloverso** [comment]
This PR hit WT-11492 which is unrelated to the changes here.

**Review by ddanderson — APPROVED** [review]
LGTM!

**Comment by ddanderson** [comment]
@sueloverso before you merge, can you run `unit-test-hook-tiered` and `unit-test-hook-tiered-s3` tasks under the `Ubuntu 20.04` variant to make sure there are no regressions.  Thanks.

File: src/cursor/cur_stat.c (hunk 1 above)
Comment (sueloverso) [reply src/cursor/cur_stat.c:771]: I recombined them and will see if it does it again. The normal combined line is 99 characters so should fit.

**Review by ddanderson — APPROVED** [review]
LGTM

**Comment by sueloverso** [comment]
@ddanderson  I'm asking for a re-review to see the changes I needed to make both to code and tests when turning on all the tiered hook tests you suggested.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

**Review by ddanderson — APPROVED** [review]
LGTM!

**Pull request merged**